src/DssSpell.t.sol

```diff
@@ -904,63 +908,346 @@ contract DssSpellTest is DssSpellTestBase {
+            uint256 amtIn = _wadToAmt(wadOut);
+            _giveTokens(address(gem), amtIn);
+            dstPsm.sellGemNoFee(address(this), amtIn);
+
+            // Buy the max amount of gems within threshold
+            uint256 wadIn = gushThreshold;
+            _giveTokens(address(dai), wadIn);
+            uint256 amtOut = _wadToAmt(wadIn);
+            dstPsm.buyGemNoFee(address(this), amtOut);
+
+            assertGt(dstPsm.gush(), 0, "no trim: invalid gush");
+            assertLt(dstPsm.gush(), gushThreshold, "no trim: invalid gush");
+
+            (bool ok, bytes memory args) = litePsmJob.workable(master);
+            assertFalse(ok, "no trim: invalid workable ok - expected false");
+            assertEq(args, bytes("No work to do"), "no trim: invalid data");
+
+            vm.expectRevert();
+            litePsmJob.work(master, args);
+
+            vm.revertTo(snapshot);
+        }
+
+        // --- chugThreshold is breached ---
+        {
+            // Donating Dai has the same effect as accumulating swap fees
+            uint256 wadDonation = cutThreshold; // Must be >= cutThreshold
+            _giveTokens(address(dai), wadDonation);
+            dai.transfer(address(dstPsm), wadDonation);
+
+            assertGe(dstPsm.cut(), cutThreshold, "chug: invalid cut");
+
+            (bool ok, bytes memory args) = litePsmJob.workable(master);
+            assertTrue(ok, "chug: invalid workable ok");
+            (bytes4 fn) = abi.decode(args, (bytes4));
+            assertEq(fn, dstPsm.chug.selector, "chug: invalid data - expected chug.selector");
+
+            litePsmJob.work(master, args);
+
+            vm.revertTo(snapshot);
+        }
+
+        // --- chugThreshold is not breached ---
+        {
+            uint256 wadDonation = cutThreshold / 5; // Must be < cutThreshold
+            _giveTokens(address(dai), wadDonation);
+            dai.transfer(address(dstPsm), wadDonation);
+
+            assertGt(dstPsm.cut(), 0, "no chug: invalid cut");
+            assertLt(dstPsm.cut(), cutThreshold, "no chug: invalid cut");
+
+            (bool ok, bytes memory args) = litePsmJob.workable(master);
+            assertFalse(ok, "no chug: invalid workable ok");
+            assertEq(args, bytes("No work to do"), "no chug: invalid data");
+
+            vm.expectRevert();
+            litePsmJob.work(master, args);
+
+            vm.revertTo(snapshot);
+        }
     }
-}
+
+    function _amtToWad(uint256 amt) internal view returns (uint256) {
+        return amt * dstPsm.to18ConversionFactor();
+    }
+
+    function _wadToAmt(uint256 wad) internal view returns (uint256) {
+        return wad / dstPsm.to18ConversionFactor();
+    }
+
+    function _min(uint256 x, uint256 y) internal pure returns (uint256 z) {
+        z = x < y ? x : y;
+    }
+
+    function _subcap(uint256 x, uint256 y) internal pure returns (uint256 z) {
+        z = x < y ? 0 : x - y;
+    }
+}
```

Comment:
Please configure your editor to always add new line in the end of the file. Some reasons [why it makes sense](https://unix.stackexchange.com/a/18789)

Reply:
fixed in [d65e60f](https://github.com/makerdao/spells-mainnet/pull/422/commits/d65e60fd5a46877932ff62a06228c6cb4b3b92fa)